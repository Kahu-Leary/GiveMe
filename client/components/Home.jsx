import React, { useEffect, useContext } from 'react'
import { isAuthenticated } from 'authenticare/client'
import { UserContext, updateUserContext } from './UserContext'

import Nav from './Nav'
// import CategoriesMenu from './CategoriesMenu'
// import GalleryImage from './GalleryImage'
import BrowseListings from './BrowseListings'
import Footer from './Footer'
function Home() {
  const [, setUser] = useContext(UserContext)

  useEffect(() => {
    if (isAuthenticated()) {
      updateUserContext(setUser)
    }
  }, [])

  return (
    <div className='container-footer'>
      <Nav />
      <BrowseListings />
      <Footer />
    </div>
  )
}

export default Home
