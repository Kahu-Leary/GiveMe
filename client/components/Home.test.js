import React from 'react'
import { HashRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { UserProvider } from './UserContext'
import Home from './Home'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HashRouter>
        <UserProvider>
          <Home/>
        </UserProvider>
      </HashRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
