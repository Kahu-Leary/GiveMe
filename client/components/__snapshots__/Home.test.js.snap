// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="globalBackground"
>
  <div
    className="container-footer"
  >
    <div
      className="alingElementNav"
    >
      <img
        className="nav-logo"
        src="./img/logo-txt-white.png"
      />
      <div
        className="alingSearchBar"
      >
        <form
          className="alingElementNav"
          onSubmit={[Function]}
        >
          <input
            className="input is-rounded is-small searchBar "
            id="searchbar"
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />
          <button
            className="button primaryBtn search"
          >
            Search
          </button>
        </form>
        <div>
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <button
                aria-controls="dropdown-menu3"
                aria-haspopup="true"
                className="button primaryBtn category"
                onClick={[Function]}
              >
                Categories
              </button>
            </div>
            <div
              className="dropdown-menu"
              id="dropdown-menu3"
              role="menu"
            >
              <div
                className="dropdown-content"
              >
                <a
                  className="dropdown-item"
                  href="#/categorylist/books"
                  onClick={[Function]}
                >
                  Books
                </a>
                <a
                  className="dropdown-item"
                  href="#/categorylist/furniture"
                  onClick={[Function]}
                >
                  Furniture
                </a>
                <a
                  className="dropdown-item"
                  href="#/categorylist/toys"
                  onClick={[Function]}
                >
                  Toys
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="navLinks"
      >
        <a
          className="link"
          href="#/signin"
          onClick={[Function]}
        >
          Add Item
        </a>
        <a
          className="link"
          href="#/Home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="link"
          href="#/About"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="link"
          href="#/ContactUs"
          onClick={[Function]}
        >
          Contact Us
        </a>
        <a
          className="link"
          href="#/SignIn"
          onClick={[Function]}
        >
          Sign In
        </a>
      </div>
    </div>
    <div>
      <ul
        className="imageGridContainer"
      />
    </div>
    <div
      className="footer"
    >
      <footer>
        <div
          className="footer-left"
        >
          <p>
            Â© 2020 GiveMe. All rights reserved
          </p>
        </div>
        <div
          className="socialMedia"
        >
          <a
            className="fa fa-facebook"
            href="#"
          />
          <a
            className="fa fa-twitter"
            href="#"
          />
          <a
            className="fa fa-google"
            href="#"
          />
          <a
            className="fa fa-github"
            href="https://github.com/pohutukawa-2020/GiveMe"
            rel="noreferrer"
            target="_blank"
          />
        </div>
      </footer>
    </div>
  </div>
</div>
`;
